src/segment-loader.js

```diff
@@ -1104,4 +1097,12 @@ export default class SegmentLoader extends videojs.EventTarget {
     this.throughput.rate +=
       (segmentProcessingThroughput - rate) / (++this.throughput.count);
   }
+
+  /**
+   * A debugging logger noop that is set to console.log only if debugging
+   * is enabled globally
+   *
+   * @private
+   */
+  logger_() {}
```

Comment:
Instead of writing this function and the check for the debug setting in each module we want to log, we should just have the actual logger passed in, and check for the debug option at the highest level.